validator/sawtooth_validator/journal/publisher_ce.py

```diff
@@ -174,8 +177,11 @@ def batches(self):
 
     def can_add_batch(self):
         return (
-            self._max_batches == 0
-            or len(self._pending_batches) < self._max_batches
+            (
+                self._max_batches == 0
+                or len(self._pending_batches) < self._max_batches
+            )
+            and self._summary is None
```

Comment:
The summary null check should go before the batch count check so that it can short-circuit before having to count the pending batches.